test/integration/utility.cc

```diff
@@ -75,27 +83,104 @@ void BufferingStreamDecoder::onResetStream(Http::StreamResetReason, absl::string
   ADD_FAILURE();
 }
 
+struct ConnectionCallbacks : public Network::ConnectionCallbacks {
+  ConnectionCallbacks(Event::Dispatcher& dispatcher) : dispatcher_(dispatcher) {}
+
+  // Network::ConnectionCallbacks
+  void onEvent(Network::ConnectionEvent event) override {
```

Comment:
RE questions asked over DM about integration_admin_test, skimming over your PR it looks like this code might be interesting for some breakpoints.

I don't have a clear picture of why it's needed, so maybe some class comments would help.